library/management/chat/src/main/java/org/quiltmc/qsl/chat/api/QuiltChatEvents.java

```diff
@@ -35,59 +36,34 @@ private QuiltChatEvents() {}
 	 * An event that allows you to modify a message before further processing by returning a new one to replace it. The usage of `withX` methods
 	 * is recommended.
 	 */
-	public static final ChatEvent<Modify, AbstractChatMessage<?>> MODIFY = new ChatEvent<>(true, (modify, quiltMessageTypes) -> modifyToHook(quiltMessageTypes, modify));
+	public static final ChatEventImpl<Modify, AbstractChatMessage<?>> MODIFY = new ChatEventImpl<>(true, InternalChatEventHandlerConverters::modifyToHook);
```

Comment:
These should all be `ChatEvent`, not `ChatEventImpl`, the interface should expose everything needed.

Reply:
```suggestion
	public static final ChatEvent<Modify, AbstractChatMessage<?>> MODIFY = new ChatEventImpl<>(true, InternalChatEventHandlerConverters::modifyToHook);
```